com.hazelcast.jet.sql.impl.parse.HazelcastSqlParserTest.test_createIndexRequiresMapName

_master_ (commit d05be9ecf18da725ac8ccfc96fe1385977cadd13)

Failed on Windows, Oracle JDK8: [test result](http://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-Windows-OracleJDK8/343/testReport/com.hazelcast.jet.sql.impl.parse/HazelcastSqlParserTest/test_createIndexRequiresMapName_true___0_/)

<details><summary>Stacktrace:</summary>

```
java.lang.AssertionError: 

Expecting throwable message:
  <"Encountered "<EOF>" at line 1, column 38.
Was expecting:
    "ON" ...
    ">
to contain:
  <"Was expecting:
    "ON" ...
">
but did not.

Throwable that failed the check:

org.apache.calcite.sql.parser.SqlParseException: Encountered "<EOF>" at line 1, column 38.
Was expecting:
    "ON" ...
    
	at com.hazelcast.jet.sql.impl.calcite.parser.HazelcastSqlParser.convertException(HazelcastSqlParser.java:412)
	at com.hazelcast.jet.sql.impl.calcite.parser.HazelcastSqlParser.normalizeException(HazelcastSqlParser.java:176)
	at org.apache.calcite.sql.parser.SqlParser.handleException(SqlParser.java:145)
	at org.apache.calcite.sql.parser.SqlParser.parseQuery(SqlParser.java:160)
	at org.apache.calcite.sql.parser.SqlParser.parseStmt(SqlParser.java:185)
	at com.hazelcast.jet.sql.impl.parse.HazelcastSqlParserTest.parse(HazelcastSqlParserTest.java:383)
	at com.hazelcast.jet.sql.impl.parse.HazelcastSqlParserTest.lambda$test_createIndexRequiresMapName$1(HazelcastSqlParserTest.java:164)
	at org.assertj.core.api.ThrowableAssert.catchThrowable(ThrowableAssert.java:62)
	at org.assertj.core.api.AssertionsForClassTypes.catchThrowable(AssertionsForClassTypes.java:750)
	at org.assertj.core.api.Assertions.catchThrowable(Assertions.java:1179)
	at org.assertj.core.api.Assertions.assertThatThrownBy(Assertions.java:1051)
	at com.hazelcast.jet.sql.impl.parse.HazelcastSqlParserTest.test_createIndexRequiresMapName(HazelcastSqlParserTest.java:164)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at junitparams.internal.InvokeParameterisedMethod.evaluate(InvokeParameterisedMethod.java:234)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at junitparams.internal.ParameterisedTestMethodRunner.runMethodInvoker(ParameterisedTestMethodRunner.java:47)
	at junitparams.internal.ParameterisedTestMethodRunner.runTestMethod(ParameterisedTestMethodRunner.java:40)
	at junitparams.internal.ParameterisedTestClassRunner.runParameterisedTest(ParameterisedTestClassRunner.java:146)
	at junitparams.JUnitParamsRunner.runChild(JUnitParamsRunner.java:446)
	at junitparams.JUnitParamsRunner.runChild(JUnitParamsRunner.java:393)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeEager(JUnitCoreWrapper.java:107)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:83)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:158)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
Caused by: com.hazelcast.jet.sql.impl.calcite.parser.ParseException: Encountered "<EOF>" at line 1, column 38.
Was expecting:
    "ON" ...
    
	at com.hazelcast.jet.sql.impl.calcite.parser.HazelcastSqlParser.generateParseException(HazelcastSqlParser.java:39435)
	at com.hazelcast.jet.sql.impl.calcite.parser.HazelcastSqlParser.jj_consume_token(HazelcastSqlParser.java:39249)
	at com.hazelcast.jet.sql.impl.calcite.parser.HazelcastSqlParser.SqlCreateIndex(HazelcastSqlParser.java:4324)
	at com.hazelcast.jet.sql.impl.calcite.parser.HazelcastSqlParser.SqlCreate(HazelcastSqlParser.java:20290)
	at com.hazelcast.jet.sql.impl.calcite.parser.HazelcastSqlParser.SqlStmt(HazelcastSqlParser.java:3408)
	at com.hazelcast.jet.sql.impl.calcite.parser.HazelcastSqlParser.SqlStmtEof(HazelcastSqlParser.java:3924)
	at com.hazelcast.jet.sql.impl.calcite.parser.HazelcastSqlParser.parseSqlStmtEof(HazelcastSqlParser.java:224)
	at org.apache.calcite.sql.parser.SqlParser.parseQuery(SqlParser.java:158)
	... 48 more

	at com.hazelcast.jet.sql.impl.parse.HazelcastSqlParserTest.test_createIndexRequiresMapName(HazelcastSqlParserTest.java:166)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at junitparams.internal.InvokeParameterisedMethod.evaluate(InvokeParameterisedMethod.java:234)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at junitparams.internal.ParameterisedTestMethodRunner.runMethodInvoker(ParameterisedTestMethodRunner.java:47)
	at junitparams.internal.ParameterisedTestMethodRunner.runTestMethod(ParameterisedTestMethodRunner.java:40)
	at junitparams.internal.ParameterisedTestClassRunner.runParameterisedTest(ParameterisedTestClassRunner.java:146)
	at junitparams.JUnitParamsRunner.runChild(JUnitParamsRunner.java:446)
	at junitparams.JUnitParamsRunner.runChild(JUnitParamsRunner.java:393)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junitcore.JUnitCore.run(JUnitCore.java:55)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.createRequestAndRun(JUnitCoreWrapper.java:137)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.executeEager(JUnitCoreWrapper.java:107)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:83)
	at org.apache.maven.surefire.junitcore.JUnitCoreWrapper.execute(JUnitCoreWrapper.java:75)
	at org.apache.maven.surefire.junitcore.JUnitCoreProvider.invoke(JUnitCoreProvider.java:158)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
```

</details>

<details><summary>Standard output:</summary>

```
14:09:29,863  INFO || - [MetricsConfigHelper] Time-limited test - [LOCAL] [dev] [5.1-SNAPSHOT] Overridden metrics configuration with system property 'hazelcast.metrics.collection.frequency'='1' -> 'MetricsConfig.collectionFrequencySeconds'='1'
14:09:29,864  INFO || - [logo] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] 
	+       +  o    o     o     o---o o----o o      o---o     o     o----o o--o--o
	+ +   + +  |    |    / \       /  |      |     /         / \    |         |   
	+ + + + +  o----o   o   o     o   o----o |    o         o   o   o----o    |   
	+ +   + +  |    |  /     \   /    |      |     \       /     \       |    |   
	+       +  o    o o       o o---o o----o o----o o---o o       o o----o    o   
14:09:29,864  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
14:09:29,864  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Hazelcast Platform 5.1-SNAPSHOT (20211113 - d05be9e) starting at [127.0.0.1]:5701
14:09:29,864  INFO || - [system] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Cluster name: dev
14:09:29,869  INFO || - [MetricsConfigHelper] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Collecting debug metrics and sending to diagnostics is enabled
14:09:29,873  WARN || - [CPSubsystem] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] CP Subsystem is not enabled. CP data structures will operate in UNSAFE mode! Please note that UNSAFE mode will not provide strong consistency guarantees.
14:09:29,876  INFO || - [JetServiceBackend] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Setting number of cooperative threads and default parallelism to 2
14:09:29,878  INFO || - [Diagnostics] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
14:09:29,878  INFO || - [LifecycleService] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] [127.0.0.1]:5701 is STARTING
14:09:29,878  INFO || - [JetExtension] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] Jet is enabled
14:09:29,878  INFO || - [ClusterService] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] 

Members {size:1, ver:1} [
	Member [127.0.0.1]:5701 - 8b0b5585-6b25-4bf0-8028-f7e79d096d58 this
]

14:09:29,878  INFO || - [LifecycleService] Time-limited test - [127.0.0.1]:5701 [dev] [5.1-SNAPSHOT] [127.0.0.1]:5701 is STARTED
Started Running Test: testNameResolution
REST API
Jenkins 2.303.1
```

</details>

--- a/hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/parse/HazelcastSqlParserTest.java
+++ b/hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/parse/HazelcastSqlParserTest.java
@@ -163,9 +163,7 @@ public class HazelcastSqlParserTest {
         // when & then
         assertThatThrownBy(() -> parse(sql))
                 .hasMessageContaining("Encountered \"<EOF>\" at line 1")
-                .hasMessageContaining(
-                        "Was expecting:\n" +
-                        "    \"ON\" ...\n");
+                .hasMessageContaining("\"ON\" ...");
     }
 
     @Test

